### DISCUSSION TITLE
Tailwind classes (text colors) not being applied to markdown

### DISCUSSION THREAD

**Discussion by hampak** [tailwindlabs/tailwindcss#12763] [Help] [answered]
Hello, I'm building a blog with NextJS, tailwindcss, and mdx. Here's the code where the error occurs:

I'm using `mdx-remote` and here's where I import all of the customizations (from code styling to custom JSX elements) which can then be applied to my mdx files.

```tsx
// mdx.tsx

import {MDXRemote} from "next-mdx-remote/rsc"

import {SomeCustomJSXComponent} from "..."

let components = {
  SomeCustomJSXComponent
}

export const MDX = (props) => {

  return (
    <>
      <MDXRemote
        {...props}
        components={{ ...components, ...(props.components || {}) }}
      />
    </>
  )
}

```

```tsx
// SomeCustomJSXComponent.tsx

export default function SomeCustomJSXComponent() {
  return (
    <div className="w-full">
      <span className="text-red-900 text-4xl">
        This is the hover Glow Effect
      </span>
    </div>
  )
}
```

```mdx
---
frontmatter goes here
---

<SomeCustomJSXComponent />
```

The problem seems to be occurring in this file:
```tsx
// SomeCustomJSXComponent.tsx

export default function SomeCustomJSXComponent() {
  return (
    <div className="w-full">
      <span className="text-red-800 text-4xl"> // class "text-red-800" doesn't get applied
        Some random text
      </span>
    </div>
  )
}
```

Tailwind classnames related to text colors doesn't seem to get applied. However, other classes such as "text-4xl" or "flex" & "justify-center" seems to be getting applied. I don't know why only classes related to text colors aren't being applied. (Please refer to the image below)

<img width="487" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/85291626/e5126914-f9ee-487a-be53-4c545906eb94">
(when everything is the same + "text-4xl" - changing text color doesn't get applied)

<img width="329" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/85291626/0378dfa4-3df5-4872-931c-5b00c7afc510">
(when everything is the same + "text-xl" - changing text color doesn't get applied)

-----

Before you ask:
- Yes, I have configured the paths in `tailwind.config.ts` correctly
- I don't know if there are other classes that don't get correctly applied other than the text color ones.

Thanks for reading, any input is welcome. I really want to know what I'm doing wrong. Please feel free to ask me any questions :D

**Comment by wongjn** [comment]
Consider providing a project that reproduces the unexpected behavior so that we can troubleshoot further.

**Comment by hampak** [comment, marked as answer]
Changing the file to `SomeCustomJSXComponent.js` instead of `SomeCustomJSXComponent.tsx` solved the entire issue! Inspired by Dan Abramov's codebase for his blog where he imported custom components.
